iframe srcdoc with quirky doctype should be no-quirks mode

rniwa opened the pull request:
#### 4401d4e5b5d9e430acd08167d20557dfcc20a569
<pre>
iframe srcdoc with quirky doctype should be no-quirks mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=159489">https://bugs.webkit.org/show_bug.cgi?id=159489</a>

Reviewed by Chris Dumez.

This patch addresses the bug that a document in srcdoc can be in quirks mode due to DOCTYPE.
New behavior of WebKit matches the spec, Gecko, and Blink.

* LayoutTests/fast/parser/strict-mode-in-srcdoc-with-quirks-doctype-expected.txt: Added.
* LayoutTests/fast/parser/strict-mode-in-srcdoc-with-quirks-doctype.html: Added.
* LayoutTests/fast/frames/srcdoc/srcdoc-can-be-in-qurks-mode-expected.txt:

* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::HTMLConstructionSite::setCompatibilityModeFromDoctype):
(WebCore::HTMLConstructionSite::insertDoctype):

Canonical link: <a href="https://commits.webkit.org/253326@main">https://commits.webkit.org/253326@main</a>
</pre>

cdumez commented on LayoutTests/fast/parser/strict-mode-in-srcdoc-with-quirks-doctype-expected.txt:
```diff
@@ -0,0 +1,14 @@
+
+PASS frame1.contentDocument.compatMode is "CSS1Compat"
+PASS frame2.contentDocument.compatMode is "CSS1Compat"
+FAIL frame3.contentDocument.compatMode should be CSS1Compat. Was BackCompat.
```
Is this expected? It doesn't seem to be explained in the test or changelog. 

I have verified that this test fully passes in Gecko & Blink. I am guessing that your patch gets us a bit closer to Gecko and Blink but still not fully aligned?

rniwa replied:
oh weird.

cdumez reviewed (approved):
LGTM assuming the bots are happy.

rniwa committed:
iframe srcdoc with quirky doctype should be no-quirks mode
https://bugs.webkit.org/show_bug.cgi?id=159489

Reviewed by Chris Dumez.

This patch addresses the bug that a document in srcdoc can be in quirks mode due to DOCTYPE.
New behavior of WebKit matches the spec, Gecko, and Blink.

* LayoutTests/fast/parser/strict-mode-in-srcdoc-with-quirks-doctype-expected.txt: Added.
* LayoutTests/fast/parser/strict-mode-in-srcdoc-with-quirks-doctype.html: Added.
* LayoutTests/fast/frames/srcdoc/srcdoc-can-be-in-qurks-mode-expected.txt:

* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::HTMLConstructionSite::setCompatibilityModeFromDoctype):
(WebCore::HTMLConstructionSite::insertDoctype):

Canonical link: https://commits.webkit.org/253326@main
